sim/hunter/item_sets_pve.go

```diff
@@ -191,3 +191,170 @@ var ItemSetGiantstalkerPursuit = core.NewItemSet(core.ItemSet{
+					if spell == hunter.RaptorStrike {
+						procAura.Activate(sim)
+					}
+				},
+			}))
+		},
+		// Increases main hand weapon damage by 5%.
+		4: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+			hunter.PseudoStats.MHDpsMultiplier += 0.05
+		},
+		// Your periodic damage has a 5% chance to reset the cooldown on one of your Strike abilities. The Strike with the longest remaining cooldown is always chosen.
+		6: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+			core.MakePermanent(hunter.RegisterAura(core.Aura{
+				Label: "S03 - Item - T2 - Hunter - Melee 6P Bonus Trigger",
+				ActionID: core.ActionID{SpellID: 467334},
+				OnPeriodicDamageDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if sim.Proc(0.05, "T2 Melee 6PC Strike Reset") {
+						maxSpell := hunter.RaptorStrike
+
+						for _, strike := range hunter.Strikes {
+							if strike.TimeToReady(sim) > maxSpell.TimeToReady(sim) {
+								maxSpell = strike
+							}
+						}
+
+						maxSpell.CD.Reset()
+						aura.Activate(sim) // used for metrics
+					}
+				},
+			}))
+		},
+	},
+})
+
+var ItemSetDragonstalkerPursuit = core.NewItemSet(core.ItemSet{
+	Name: "Dragonstalker's Pursuit",
+	Bonuses: map[int32]core.ApplyEffect{
+		// Your Aimed Shot deals 20% more damage to targets afflicted by one of your trap effects.
+		2: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+
+			procAura := hunter.RegisterAura(core.Aura{
+				Label: "S03 - Item - T2 - Hunter - Ranged 2P Bonus",
+				ActionID: core.ActionID{SpellID: 467235},
+				Duration: time.Second * 1, // Duration doesn't really matter as the duration is set to the duration of the trap debuff that applies this aura
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+					hunter.AimedShot.DamageMultiplier *= 1.20
+				},
+				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+					hunter.AimedShot.DamageMultiplier /= 1.20
+				},
+			})
+			
+			core.MakePermanent(hunter.RegisterAura(core.Aura{
+				Label: "S03 - Item - T2 - Hunter - Ranged 2P Bonus Trigger",
+				OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if spell.Flags.Matches(SpellFlagTrap) {
+						procAura.Duration = spell.DotOrAOEDot(result.Target).Duration
+						procAura.Activate(sim)
+					}
+				},
+			}))
+		},
+		// Your damaging Shot abilities deal 10% increased damage if the previous damaging Shot used was different than the current one.
+		4: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+
+			procAura := hunter.RegisterAura(core.Aura{
+				ActionID: core.ActionID{SpellID: 467312},
+				Label:    "S03 - Item - T2 - Hunter - Ranged 4P Bonus",
+				Duration: time.Second * 12,
+
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+					for _, spell := range hunter.Shots {
+						if spell != nil && spell != hunter.LastShot {
```

Comment:
You might be able to not check for not nil here, and then you may also want to check spell codes instead of just the spell struct. You would need to also check for a `nil` LastShot before checking SpellCode though